### PR TITLE
[CherryPick] Lucene search IMP

### PR DESCRIPTION
**Pull request by QilongTang**
### Purpose

Cherry-picking:
#14428
https://github.com/DynamoDS/Dynamo/pull/14513

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

N/A

### FYIs

@DynamoDS/dynamo

### CODE REVIEW

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs [hunk 1]
```diff
@@ -64,4 +64,4 @@
 // You can specify all the values or you can default the Build and Revision Numbers
 // by using the '*' as shown below:
 // [assembly: AssemblyVersion("1.0.*")]
-[assembly: AssemblyFileVersion("2.19.5.7897")]
```
Comment (RobertGlobant20) [inline src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs:67]: it's ok to modify this file?

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs [hunk 2]
```diff
@@ -45,7 +45,7 @@
 // to distinguish one build from another. AssemblyFileVersion is specified
 // in AssemblyVersionInfo.cs so that it can be easily incremented by the
 // automated build process.
-[assembly: AssemblyVersion("2.19.5.7897")]
+[assembly: AssemblyVersion("2.19.5.7936")]
```
Comment (reddyashish) [inline src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs:48]: Remove this file commit.

**Review by reddyashish — COMMENTED** [review]
LGTM with one comment.

**Review by RobertGlobant20 — APPROVED** [review]
LGTM, this are changes already applied in 3.0, right? This is just a cherry-pick

**Pull request merged**